http-api/src/lib.rs

```diff
@@ -247,15 +265,58 @@ async fn blob_handler(
     Ok(warp::reply::json(&blob))
 }
 
-async fn commits_handler(
+async fn remotes_handler(ctx: Context, project: Urn) -> Result<impl Reply, Rejection> {
+    let storage = ReadOnly::open(&ctx.paths).map_err(Error::from)?;
+    let tracked_peers = tracking::tracked(&storage, &project).map_err(|_| Error::NotFound)?;
+    let response = tracked_peers.collect::<Vec<_>>();
+
+    Ok(warp::reply::json(&response))
+}
+
+async fn remote_handler(
     ctx: Context,
     project: Urn,
     peer_id: PeerId,
-    sha: One,
 ) -> Result<impl Reply, Rejection> {
-    let reference = Reference::head(Namespace::from(project), peer_id, sha);
-    let commits = browse(reference, ctx.paths, |browser| {
-        radicle_source::commits::<PeerId>(browser, None)
+    let meta = get_project_metadata(&project, &ctx.paths).map_err(Error::from)?;
+
+    let repo = git::Repository::new(ctx.paths.git_dir().to_owned()).map_err(Error::from)?;
+    let mut browser = git::Browser::new_with_namespace(
+        &repo,
+        &git::Namespace::try_from(project.encode_id().as_str()).map_err(Error::from)?,
+        get_remote_head_branch(&meta.default_branch, &peer_id).map_err(Error::from)?,
+    )
+    .map_err(Error::from)?;
+
+    let branches = browser
+        .list_branches(git::RefScope::Remote {
+            name: Some(peer_id.default_encoding()),
+        })
+        .map_err(Error::from)?
+        .iter()
+        .filter_map(
+            |branch| match !branch.name.to_string().starts_with("rad/") {
+                true => {
+                    browser
+                        .branch(get_remote_head_branch(&branch.name.to_string(), &peer_id).ok()?)
+                        .unwrap();
```

Comment:
I'm having an issue with using the `?` operator and/or `map_err` since I get the following
```
the` `?` operator can only be used on `Option`s, not `Result`s, in a closure that returns `Option`
```
The solution I see is to discard the error information with eg. `.ok()?` or `into_ok()` would also be great, but is on nightly..

Reply:
I think I found a good compromise.
There is the following error that doesn't allow me to just use map_err
```
the `?` operator can only be used on `Option`s, not `Result`s, in a closure that returns `Option`
```